### ISSUE TITLE
Inconsistent specifications for HDUCLAS2 keyword in EVENTS extension

### ISSUE THREAD

**Issue by lmohrmann** [open-gamma-ray-astro/gamma-astro-data-formats#91]
The specifications for the HDUCLAS2 header keyword in the EVENTS extension are inconsistent: HDUCLAS2 is listed as a mandatory (!) header keyword [here](http://gamma-astro-data-formats.readthedocs.io/en/latest/events/events.html), but not specified at all for the EVENTS extension [here](http://gamma-astro-data-formats.readthedocs.io/en/latest/general/hduclass.html).

**Comment by cdeil** [comment]
The `HDUCLAS2` for `EVENTS` was introduced by @jknodlseder [here](https://github.com/open-gamma-ray-astro/gamma-astro-data-formats/commit/68c9aaf6cf71ad74eea11552e98158e766433d96#diff-ab2e58774ff2e0930caa191b5c3a8ecfR119) and then that bullet was slightly modified by @TarekHC (I think, not sure if that is your commit) [here](https://github.com/open-gamma-ray-astro/gamma-astro-data-formats/commit/a8c089d30989f2d2815a834a1e3b76ea9091a4a6#diff-ab2e58774ff2e0930caa191b5c3a8ecfR111).

This key is mentioned here in this OGIP recommendation, and they don't make the distinction optional / required like we do:
https://heasarc.gsfc.nasa.gov/docs/heasarc/ofwg/docs/ofwg_recomm/hduclas.html

I don't think science tools currently read and use this keyword, so my suggestion would be to change it to optional and not mention it in the summary table. But I don't care much either way how this detail is changed.

@jknodlseder @TarekHC - Thoughts?

**Comment by TarekHC** [comment]
> I don't think science tools currently read and use this keyword, so my suggestion would be to change it to optional and not mention it in the summary table. But I don't care much either way how this detail is changed.

I'm not sure if I agree here. Data formats should be as stable as possible in time (so we may not used them now, but maybe in the future), and having some extra keywords formally describing each table class makes sense. I would personally make them mandatory, even if for the moment they are not used (but may help astronomers understand HDUs a bit better...).

**Comment by lmohrmann** [comment]
Then the table on [this page](http://gamma-astro-data-formats.readthedocs.io/en/latest/general/hduclass.html) should maybe be updated? Are there other suggested/possible values for HDUCLAS2 other than ACCEPTED?

**Comment by cdeil** [comment]
The OGIP spec (see https://heasarc.gsfc.nasa.gov/docs/heasarc/ofwg/docs/ofwg_recomm/hduclas.html) mentions these values:
```
n=1 | 2 | 3 | COMMENTS
-- | -- | -- | --
EVENTS |   |   | Conforms to format defined in "The Proposed Timing FITS File Format", Angelini, Pence and Tennant (1993), and "The Recommended Columns and Keywords for a FITS Event List", Pence and George (1995, Legacy 6, 36)
  | ALL |   | Photon Event List, includes all photons detected without screening
  | ACCEPTED |   | Only accepted photons included in table
  | REJECTED |   | Only rejected photons included in table
```

I see that Fermi-LAT puts this key and "ALL", even after applying background suppression and other selections.

I would say for us at DL3, there have always been some selections and "ACCEPTED" fits best.

@TarekHC - The fact that it doesn't matter what we put here is another indication that the key should be optional. Note that if we make it required, then all event lists produced so far (for the past 5 years) are invalid under our spec. IMO the required keys should be the minimal subset that's useful for analysis.

---

@lmohrmann or anyone willing to make a pull request -- I'd be happy with any changes to the spec to wrap up this discussion for now.

**Comment by TarekHC** [comment]
> @TarekHC - The fact that it doesn't matter what we put here is another indication that the key should be optional. 

Fine by me, I have no strong opinion. I like declarative schemes, they may be very useful for people starting to play with DL3 data. But suggesting their use may be good enough for now.

**Comment by lmohrmann** [comment]
I just modified pull request #92 to include this change (i.e. to make HDUCLAS2 an optional header keyword).

**Comment by cdeil** [comment]
I think this is fully addressed in #92 ? Closing now.

@lmohrmann - Thanks!
